### PR TITLE
Cache some settings

### PR DESCRIPTION
**Pull request by rubenwardy**
Removes 2,500 setting calls from server tick (according to my beginner usage of callgrind).

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -670,20 +670,21 @@ void Server::AsyncRunStep(bool initial_step)
...
-				!g_settings->getBool("unlimited_player_transfer_distance"))
+		static const s16 radius =
+			g_settings->getS16("active_object_send_range_blocks") * MAP_BLOCKSIZE;
+
+		// Radius inside which players are active
+		static const bool is_transfer_limited =
+			g_settings->exists("unlimited_player_transfer_distance") &&
+			!g_settings->getBool("unlimited_player_transfer_distance");
+		s16 player_radius =
+			g_settings->getS16("player_transfer_distance") * MAP_BLOCKSIZE;
```
Comment (ShadowNinja) [inline src/server.cpp]: Why isn't this cached?

**Comment by ShadowNinja** [comment]
:+1:

File: src/server.cpp (hunk 1 above)
Comment (rubenwardy) [inline src/server.cpp]: It's potentially a useful feature. Admittedly it should be implemented a different way - some per-player minetest.set_nametag_attributes call.

Should I cache it?

Comment (ShadowNinja) [inline src/server.cpp]: I think that if you're caching `unlimited_player_transfer_distance` you have to cache the distance too.

Comment (rubenwardy) [inline src/server.cpp]: Unlimited_player_transfer_distance is a depreciated setting. It's replaced
be player_transfer_distance.

Comment (ShadowNinja) [inline src/server.cpp]: Ah, O.K.  I don't think the player transfer distance is likely to be changed much though, so I still think it should be cached.

**Comment by MoNTE48** [comment]
:+1: Why it has not yet been added?

**Comment by paramat** [comment]
Needs 2nd reviewer.

**Comment by paramat** [comment]
@rubenwardy code looks good to me but what about ShadowNinja's suggestion about cacheing 'player radius'?

**Comment by est31** [comment]
:+1: looks good

**Comment by rubenwardy** [comment]
Rebased and improvements made. Testing now.

**Comment by rubenwardy** [comment]
Compile error fixed. Ready for reviews.

**Comment by paramat** [comment]
:+1:

**Comment by celeron55** [comment]
Maybe these settings are changed rarely enough to warrant a temporary optimization like this but I really wish there was some way of updating these without restarting the server. This is ultimately just an another kind of compromise.

This isn't documented either; nobody knows which settings can be modified at runtime and which can not.

**Comment by est31** [comment]
I agree, optimally we'd speed up the settings system to a simple array look up, then this PR isn't required.

**Comment by rubenwardy** [comment]
What happened to the enum based one? It could be a good idea to have basic engine settings as enums which are then used to look up in an array.

**Comment by rubenwardy** [comment]
#2695

**Comment by paramat** [comment]
4efb7eb7347bfd712297f5e11d3a055bb74cf597

**Pull request closed without merging**